show_title

paul07iq opened the pull request:
Added show_title function

paul07iq committed:
show_title

added "show_title" function for player

paul07iq committed:
Added reset_title/clear_title

Added reset_title/clear_title

paul07iq committed:
Added show_title, clear_title, reset_title

Added show_title, clear_title, reset_title

paul07iq committed:
fixed show_title

fixed show_title

paul07iq committed:
fixed show_title

fixed show_title

paul07iq committed:
fixed color at show_title

fixed color at show_title

paul07iq committed:
fixed color at show_title

fixed color at show_title

paul07iq committed:
fixed show_title

fixed show_title

paul07iq committed:
fixed show_title

fixed show_title

paul07iq committed:
fixed import

fixed import

paul07iq committed:
fixed time

fixed time

paul07iq committed:
fixed time

fixed time

paul07iq committed:
fixed show_title

fixed show_title

paul07iq committed:
fixed show_title

fixed show_title

paul07iq committed:
fixed show_title

fixed show_title

paul07iq committed:
fixed show_title

fixed show_title

paul07iq committed:
fixed color at show_title

fixed color at show_title

paul07iq committed:
fixed color at show_title

fixed color at show_title

paul07iq committed:
fixed color at show_title

fixed color at show_title

paul07iq committed:
fixed port

fixed port

paul07iq committed:
Improved Color at show_title

Improved Color at show_title

paul07iq committed:
Improved Color at show_title

Improved Color at show_title

paul07iq committed:
Added slots to _types.py

Added slots to _types.py

paul07iq committed:
Removed slots

Removed slots

icezyclon commented on mcproto/minecraft.py:
```diff
@@ -54,3 +55,13 @@ def port(self) -> int:
         response = self._stub.postToChat(pb.ChatPostRequest(message=sep.join(map(str, args))))
         raise_on_error(response)
+
+    def show_title(self, text: str, typ: Literal["actionbar", "subtitle", "title"] = "title", color: COLOR = "gray", bold: bool = "false", italic: bool = "false", strikethrough: bool = "false", underlined: bool = "false", obfuscated: bool = "false", duration: int = 1, fade_in: int = 5, fade_out: int = 1) -> None:
```
Bitte gleiche Schreibweise von Funktionen weiterverwenden, wie im Projekt. Alle anderen Funktionen verwenden camelCase nicht snake_case.

Hier gibt es Typ-Unstimmigkeiten:
Du sagst zwar dass die Variablen bold, italic, usw. "bool" sind, sie sind aber "str" hier. Bitte auf echte "bool" ändern und falls Groß-Kleinschreibung im Befehl eine Rolle spielt, mit "str(True/False).lower()" umwandeln.

Zuletzt wäre ein Vorschlag die Duration auf 5 und den FadeIn auf 1 zu setzten.

icezyclon commented on mcproto/minecraft.py:
```diff
@@ -5,6 +5,7 @@
 
```
Zunächst ist der Typ "Literal" nicht bekannt in diesem File. Der Literal Typ wird nur zufällig von einem anderen Modul importiert, sollte aber explizit importiert werden.
Hier also bitte "from typing import Literal"

icezyclon commented on mcproto/minecraft.py:
```diff
@@ -54,3 +55,13 @@ def port(self) -> int:
+        HasStub.runCommand(self, f'title @a times {fade_in}s {duration}s {fade_out}s')
+        HasStub.runCommand(self, f'title @a {typ} ' + '{' + f'"text":"{text}","color":"{color}","bold":{bold},"italic":{italic},"strikethrough":{strikethrough},"underlined":{underlined},"obfuscated":{obfuscated}' + '}')
+
+    def reset_title(self):
```
Wozu brauchen wir resetTitle? Wird nicht eh jedes Mal, wenn wir showTitle aufrufen, die Attribute neu gesetzt?

icezyclon commented on mcproto/minecraft.py:
```diff
@@ -54,3 +55,13 @@ def port(self) -> int:
+        HasStub.runCommand(self, f'title @a reset')
+        
+    def clear_title(self):
+        HasStub.runCommand(self, f'title @a clear')
```
Braucht kein f-string sein.

icezyclon commented on mcproto/player.py:
```diff
@@ -118,6 +119,16 @@ def op(self) -> None:
     def deop(self) -> None:
         HasStub.runCommand(self, f"deop {self.name}")
 
+    def show_title(self, text: str, typ: Literal["actionbar", "subtitle", "title"] = "title", color: COLOR = "gray", bold: bool = "false", italic: bool = "false", strikethrough: bool = "false", underlined: bool = "false", obfuscated: bool = "false", duration: int = 1, fade_in: int = 5, fade_out: int = 1) -> None:
```
Bitte gleiche Änderungen wie oben für die Typen und den Case.

icezyclon commented on mcproto/player.py:
```diff
@@ -118,6 +119,16 @@ def op(self) -> None:
 
+    def show_title(self, text: str, typ: Literal["actionbar", "subtitle", "title"] = "title", color: COLOR = "gray", bold: bool = "false", italic: bool = "false", strikethrough: bool = "false", underlined: bool = "false", obfuscated: bool = "false", duration: int = 1, fade_in: int = 5, fade_out: int = 1) -> None:
+        HasStub.runCommand(self, f'title {self.name} times {fade_in}s {duration}s {fade_out}s')
+        HasStub.runCommand(self, f'title {self.name} {typ} ' + '{' + f'"text":"{text}","color":"{color}","bold":{bold},"italic":{italic},"strikethrough":{strikethrough},"underlined":{underlined},"obfuscated":{obfuscated}' + '}')
```
Da wir in der Klasse Player > Entity hier sind, können wir ruhig die überschriebene "runCommand" Funktion dieser Klasse verwenden. Hier direkt HasStub.runCommand zu verwenden umgeht die Klassenhierachie.
Wenn man das macht, könnte man auch bei allen Befehlen einfach 'title @s ...' schreiben, da der Player identifier schon in Entity.runCommand gesetzt wird :+1:

icezyclon replied:
Bitte diesen Punkt noch ausbessern, alle Instanzen von `HasStub.runCommand` durch `self.runCommand` ersetzten, und im Falle von der Player-Klasse, mit `title @s ...` anstelle von `title {self.name} ...`

icezyclon reviewed (changes requested):
Code funktioniert insofern ich das testen konnte, es gibt einige Typ-Unstimmigkeiten und Details, die mit Projektstruktur zu tun haben, ansonsten ein sehr guter erster Commit.
Ich würde den Commit akzeptieren, wenn die angemerkten Änderungen durchgeführt werden. Gute Arbeit!

paul07iq committed:
Update minecraft.py

- Used camelCase
- Removed reset_title
- Change default values
- Imported Literal

paul07iq committed:
Update player.py

- Used camelCase
- Removed reset_title
- Change default values

icezyclon commented on mcproto/minecraft.py:
```diff
@@ -54,3 +56,10 @@ def port(self) -> int:
         response = self._stub.postToChat(pb.ChatPostRequest(message=sep.join(map(str, args))))
         raise_on_error(response)
+
+    def showTitle(self, text: str, typ: Literal["actionbar", "subtitle", "title"] = "title", color: COLOR = "gray", bold: bool = "false", italic: bool = "false", strikethrough: bool = "false", underlined: bool = "false", obfuscated: bool = "false", duration: int = 3, fade_in: int = 1, fade_out: int = 1) -> None:
```
Und zuletzt bitte noch die Datentypen anpassen: Datentyp `bool`sollte als default Argument True oder False haben, nicht "true" oder "false". :+1:

icezyclon reviewed (changes requested):
Danke fürs ausbessern soweit. Bitte noch die zwei letzten Kommentare zu 1) Datentypen und 2) HasStub -> self ersetzten, dann sind alle Punkte erledigt :+1:

icezyclon commented:
Übrigens, es gibt noch einen Punkt der wichtig für mich ist: und zwar, gibt es ein default formatting in diesem Projekt, das mit `pre-commit` gemacht wird. Die passende Konfiguration dafür liegt im Repository.
Bevor du einen PR einreichst, würde ich dich bitte sowohl `requirements.txt` als auch `requirements-dev.txt` zu installieren, und im Anschluss `pre-commit install` auszuführen, damit pre-commit installiert wird. (Es scheint auch ein Plugin für PyCharm zu geben: https://plugins.jetbrains.com/plugin/9278-pre-commit-hook-plugin)
Dieses wird dann jedes Mal wenn du einen Commit machst, alle Dateien formatieren, sodass alle Dateien im Repository gleich aussehen :+1:

Auf diese Art und Weise ist alles einheitlich und sortiert.

paul07iq committed:
Update minecraft.py

- Duration Standartwert auf 5s erhöht
- HasStub.runCommand durch self.runCommand ersetzt

paul07iq committed:
Update player.py

- Duration Standartwert auf 5s erhöht
